src/main/kotlin/snyk/container/KubernetesImageCache.kt

```diff
@@ -0,0 +1,87 @@
+package snyk.container
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.progress.runBackgroundableTask
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.newvfs.events.VFileContentChangeEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileCopyEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileCreateEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileDeleteEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileMoveEvent
+import com.jetbrains.rd.util.concurrentMapOf
+import org.jetbrains.yaml.YAMLFileType
+
+object KubernetesImageCache {
+    private val images = concurrentMapOf<VirtualFile, Set<KubernetesWorkloadImage>>()
+    internal val logger = Logger.getInstance(javaClass.name)
+
+    fun clear() {
+        images.clear()
+    }
+
+    fun scanProjectForKubernetesFiles(project: Project) {
+        val title = "Snyk: Scanning For Kubernetes Files..."
```

Comment:
should proceed only if Container enabled

Reply:
In this story there is no feature flag - once it's there, it could be asked. Have you added one?

Reply:
added flag to avoid running it in prod